game/gbtd_gbr0.ksy

```diff
@@ -0,0 +1,230 @@
+meta:
+  id: gbr0
+  endian: le
+  encoding: ascii
+  title: GBR file
+  license: MIT
+  application: GameBoy Tile Designer, GBTD
+  file-extension:
+    - gbr
+  tags:
+    - game
+
+doc: |
+  GBR0 files are used by GameBoy Tile Designer
+  to store tilemaps created with the program.
+doc-ref: https://www.devrs.com/gb/hmgd/gbtd.html
+
+seq:
+  - id: magic
+    contents: GBO0
+  - id: objects
+    type: gbr0_object
+    repeat: eos
+
+types:
+  gbr0_object:
+    seq:
+      - id: object_type
+        type: u2
+        enum: type
+      - id: object_id
+        type: u2
+      - id: record_length
+        type: u4
+      - id: body
+        size: record_length
+        type:
+          switch-on: object_type
+          cases:
+            type::producer:      producer
+            type::tile_data:     tile_data
+            type::tile_settings: tile_settings
+            type::tile_export:   tile_export
+            type::tile_import:   tile_import
+            type::palettes:      palettes
+            type::tile_pal:      tile_pal
+            type::deleted:       deleted
+    
+    enums:
+      type:
+        0x0001: producer
+        0x0002: tile_data
+        0x0003: tile_settings
+        0x0004: tile_export
+        0x0005: tile_import
+        0x000D: palettes
+        0x000E: tile_pal
+        0x00FF: deleted
+
+    types:
+      producer:
+        seq:
+          - id: name
+            type: strz
+            size: 30
+          - id: version
+            type: strz
+            size: 10
+          - id: info
+            type: strz
+            size: 80
+
+      tile_data:
+        seq:
+          - id: name
+            type: strz
+            size: 30
+          - id: width
+            type: u2
+          - id: height
+            type: u2
+          - id: count
+            type: u2
+          - id: color_set
+            size: 4
+          - id: tiles
+            size: width*height
+            repeat: expr
+            repeat-expr: count
+
+      tile_settings:
+        seq:
+          - id: tile_id
+            type: u2
+          - id: simple
+            type: u1
+          - id: flags
```

Comment:
We can specify fields of size in bits using `type: b<size in bits here>`, i.e. `b1` for single bit. May make sense to have a type `flags` with such bit-fields.

Reply:
Will fix